pkg/pipeline/lifecycle.go

```diff
@@ -528,28 +550,32 @@ func (s *Service) runPipeline(ctx context.Context, pl *Instance) error {
 		return cerrors.Errorf("pipeline not updated: %w", err)
 	}
 
-	// We're using different tombs for the pipeline and the nodes,
-	// since we need to be sure that all nodes are stopped,
-	// before declaring the pipeline as stopped.
-	pl.t = &tomb.Tomb{}
+	// cleanup function updates the metrics and pipeline status once all nodes
+	// stop running
 	pl.t.Go(func() error {
-		//nolint:staticcheck // nil used to use the default (parent provided via WithContext)
-		ctx := pl.t.Context(nil)
-		err := nodesTomb.Wait()
+		// use fresh context for cleanup function, otherwise the updated status
+		// won't be stored
+		ctx := context.Background()
+
+		nodesWg.Wait()
+		err := pl.t.Err()
 
 		measure.PipelinesGauge.WithValues(strings.ToLower(pl.Status.String())).Dec()
 
-		switch {
-		case cerrors.Is(err, ErrGracefulShutdown):
-			// not an actual error, it was a graceful shutdown, that is expected
+		if err == tomb.ErrStillAlive {
```

Comment:
I think this check would be nicer as a switch, instead of nested if-else statements so that it would be a switch statement with if-else inside

Reply:
Nice suggestion, thanks 👍